sqlmesh/core/config/linter.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlmesh.core.config.base import BaseConfig
+
+
+class LinterConfig(BaseConfig):
+    """Configuration for model linting
+
+    Args:
+        enabled: Flag indicating whether the linter should run
+
+        rules: A list of error rules to be applied on model
+        warn_rules: A list of rules to be applied on models but produce warnings instead of raising errors.
+        exclude_rules: A list of rules to be excluded/ignored
+
+    """
+
+    enabled: bool = False
+
+    rules: t.Union[t.List[str], str] = []
+    warn_rules: t.Union[t.List[str], str] = []
+    exclude_rules: t.Union[t.List[str], str] = []
```

Comment:
I actually think the type alias made sense here because `str` can't be a generic string, it's just `"ALL"`, so we should reflect this using `Literal["ALL"]`.

cc @tobymao did you have anything else in mind [here](https://github.com/TobikoData/sqlmesh/pull/3787/files/9513d23551d379c8ad05562838feda728716f41b#r1973890986)?

Reply:
can we just do warn_rules t.List[str] and get rid of the union, if one of the strings is ALL, we're done